jenkins/PerfCI_OpenShift_Deployment/Jenkinsfile

```diff
@@ -0,0 +1,114 @@
+pipeline {
+    agent {
+        label 'PerfCi'
+    }
+    environment {
+        CONTACT_EMAIL1 = credentials('perfci_contact_email1')
+        PROVISION_PRIVATE_KEY_FILE = credentials('perfci_provision_private_key_file')
+        PROVISION_IP = credentials('perfci_provision_ip')
+        PROVISION_USER = credentials('perfci_provision_user')
+        WORKSPACE = '/home/jenkins'
+        INSTALL_OCP_VERSION = 'latest-4.14'
+        OCP_CLIENT_VERSION = '4.14.1'
+        PROVISION_PORT = 22
+        KUBEADMIN_PASSWORD_PATH = '/root/.kube/kubeadmin-password'
+        KUBECONFIG_PATH = '/root/.kube/config'
+        PROVISION_INSTALLER_PATH = '/root/jetlag/./run_jetlag.sh'
+        PROVISION_INSTALLER_CMD = 'pushd /root/jetlag;/root/jetlag/./run_jetlag.sh 1>/dev/null 2>&1;popd'
+        PROVISION_INSTALLER_LOG = 'tail -40 /root/jetlag/jetlag.log'
+        PRIVATE_KEY_PATH = '/home/jenkins/.ssh/provision_private_key'
+        CONFIG_PATH = '/home/jenkins/.ssh/config'
+        CONTAINER_PRIVATE_KEY_PATH = '/root/.ssh/provision_private_key'
+        CONTAINER_CONFIG_PATH = '/root/.ssh/config'
+        PROVISION_TIMEOUT = 7200
+    }
+    stages {
+        stage('Cleanup') {
+            steps {
+                script {
+                    sh 'sudo podman ps -a --filter "name=benchmark-runner" -q | xargs -r podman rm'
+                }
+            }
+        }
+
+        stage('⚙️ SET SSH key') {
+            steps {
+                script {
+                    sh 'mkdir -p $WORKSPACE/.ssh/'
+                    withCredentials([file(credentialsId: 'perfci_provision_private_key_file', variable: 'PROVISION_PRIVATE_KEY_FILE')]) {
+                        sh "sudo cp \$PROVISION_PRIVATE_KEY_FILE $WORKSPACE/.ssh/provision_private_key"
```

Comment:
*This* needs to run under `umask 77` rather than a later chmod.  This is a private value that you do not want even momentarily exposed.

Reply:
Got the following error without sudo: 
`ERRO[0000] XDG_RUNTIME_DIR directory "/run/user/0" is not owned by the current user`

Reply:
Yes, but if somebody manages to open the file in the (short) window between when you create it and when you chmod it, they could grab the value.  Make sure that you use `umask 77` while you create directories and files rather than relying on chmod after the fact.

Reply:
I dont see any impact to use umask 77, I think its more clear to run chmod 600 after

Reply:
If you use `umask 77`, any files created will already be mode 600, so you don't need the explicit chmod.

Reply:
ok so I will fix it to:
`umask 77 && mkdir -p $WORKSPACE/.ssh/`

Reply:
Since the directory is created under umask 77 it will never be accessible, so it's OK for the private key underneath it to be not readable.  Best practice would be to use umask 77 for that too, but this is acceptable.